worker/handler/handler.go

```diff
@@ -106,24 +113,72 @@ func NewHandlerManagerSet(opts *config.Config) (hms *HandlerManagerSet, err erro
 	var ihits int64 = 0
 	var misses int64 = 0
 	hms = &HandlerManagerSet{
-		hmMap:      make(map[string]*HandlerManager),
-		regMgr:     rm,
-		pipMgr:     pm,
-		sbFactory:  sf,
-		cacheMgr:   cm,
-		config:     opts,
-		workerDir:  opts.Worker_dir,
-		maxRunners: opts.Max_runners,
-		hhits:      &hhits,
-		ihits:      &ihits,
-		misses:     &misses,
+		hmMap:              make(map[string]*HandlerManager),
+		regMgr:             rm,
+		pipMgr:             pm,
+		sbFactory:          sf,
+		cacheMgr:           cm,
+		config:             opts,
+		workerDir:          opts.Worker_dir,
+		maxRunners:         opts.Max_runners,
+		hhits:              &hhits,
+		ihits:              &ihits,
+		misses:             &misses,
+		avgHandlerMemUsage: opts.Average_handler_memory_usage * 1024,
+		sbHardLimit:        opts.Create_sandbox_hard_limit * 1024,
+		numSbCreating:      0,
 	}
 
+	hms.sbCreatingChan = make(chan bool, 1)
+
 	hms.lru = NewHandlerLRU(hms, opts.Handler_cache_size) //kb
 
+	if hms.avgHandlerMemUsage == 0 {
+		go func() {
```

Comment:
There's too much complexity here to have it be an anonymous function.  Let's move it out to it's own function.

Reply:
Updated in the new commit.